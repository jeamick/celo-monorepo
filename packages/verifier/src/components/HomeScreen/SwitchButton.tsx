import colors from '@celo/react-components/styles/colors'
import * as React from 'react'
import { StyleSheet, Switch, TextStyle, View } from 'react-native'

interface SwitchButtonProps {
  style?: TextStyle
  switchStatus: boolean
  onToggle: () => void
}

class SwitchButton extends React.Component<SwitchButtonProps> {
  render() {
    const { onToggle, switchStatus, style } = this.props
    return (
      <View style={[styles.switchContainer, style]}>
        <Switch
          style={styles.switch}
          thumbColor={colors.white}
          trackColor={{ true: colors.celoGreen, false: colors.darkSecondary }}
          onValueChange={onToggle}
          value={switchStatus}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  switchContainer: {
    alignContent: 'center',
    justifyContent: 'center',
  },
  switch: {
    padding: 5,
  },
})

export default SwitchButton
