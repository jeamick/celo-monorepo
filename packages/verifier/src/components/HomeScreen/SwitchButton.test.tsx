import * as React from 'react'
import 'react-native'
import * as renderer from 'react-test-renderer'
import SwitchButton from 'src/components/HomeScreen/SwitchButton'

it('renders correctly', () => {
  const tree = renderer.create(<SwitchButton onToggle={jest.fn()} switchStatus={false} />)
  expect(tree).toMatchSnapshot()
})
