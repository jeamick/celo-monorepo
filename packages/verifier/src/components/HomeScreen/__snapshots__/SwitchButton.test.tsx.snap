// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <RCTSwitch
    accessibilityRole="button"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="#42D689"
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Object {
          "padding": 5,
        },
      ]
    }
    thumbTintColor="#FFFFFF"
    tintColor="#81868B"
    value={false}
  />
</View>
`;
